### ISSUE TITLE
Unify Modals

### ISSUE THREAD

**Issue by Gregoor** [mozilla/voice-web#593]
Create a Modal component, that replaces `confirm.tsx` and `.overlay` (used for Privacy). It would also be used for the Contact Form and Keep/Delete Data Modals from the new designs.

To not reinvent the wheel (and decrease the amount of code we have to maintain), I'd like to go with `react-modal`. Unfortunately `preact` can't keep its API-compatibility promise here, especially not for React 16, so it doesn't work with preact (to be fair, for React<15 it uses a React-interal API). This makes the switch to React 16 even more attractive, which is why this issue kind of depends on #591

**Comment by Gregoor** [comment]
Oh, overlooked #234. Closing this as it is a duplicate of it.
